---a/net/minecraft/world/biome/BiomeGenSavanna.java
+++b/net/minecraft/world/biome/BiomeGenSavanna.java
@@ -1,6 +1,7 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDirt;
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.entity.passive.EntityHorse;
@@ -11,13 +11,13 @@
 import net.minecraft.world.gen.feature.WorldGenAbstractTree;
 import net.minecraft.world.gen.feature.WorldGenSavannaTree;
 
-public class BiomeGenSavanna extends BiomeGenBase
-{
+public class BiomeGenSavanna extends BiomeGenBase {
+
     private static final WorldGenSavannaTree field_150627_aC = new WorldGenSavannaTree(false);
     private static final String __OBFID = "CL_00000182";
 
-    protected BiomeGenSavanna(int p_i45383_1_)
-    {
+    public BiomeGenSavanna(int p_i45383_1_) {
         super(p_i45383_1_);
         this.spawnableCreatureList.add(new BiomeGenBase.SpawnListEntry(EntityHorse.class, 1, 2, 6));
         this.theBiomeDecorator.treesPerChunk = 1;
@@ -25,14 +25,12 @@
         this.theBiomeDecorator.grassPerChunk = 20;
     }
 
-    public WorldGenAbstractTree genBigTreeChance(Random p_150567_1_)
-    {
-        return (WorldGenAbstractTree)(p_150567_1_.nextInt(5) > 0 ? field_150627_aC : this.worldGeneratorTrees);
+    public WorldGenAbstractTree genBigTreeChance(Random p_150567_1_) {
+        return (WorldGenAbstractTree) (p_150567_1_.nextInt(5) > 0 ? field_150627_aC : this.worldGeneratorTrees);
     }
 
-    protected BiomeGenBase createMutatedBiome(int p_180277_1_)
-    {
+    public BiomeGenBase createMutatedBiome(int p_180277_1_) {
         BiomeGenSavanna.Mutated mutated = new BiomeGenSavanna.Mutated(p_180277_1_, this);
         mutated.temperature = (this.temperature + 1.0F) * 0.5F;
         mutated.minHeight = this.minHeight * 0.5F + 0.3F;
@@ -39,12 +39,10 @@
         return mutated;
     }
 
-    public void decorate(World worldIn, Random p_180624_2_, BlockPos p_180624_3_)
-    {
+    public void decorate(World worldIn, Random p_180624_2_, BlockPos p_180624_3_) {
         DOUBLE_PLANT_GENERATOR.func_180710_a(BlockDoublePlant.EnumPlantType.GRASS);
 
-        for (int i = 0; i < 7; ++i)
-        {
+        for (int i = 0; i < 7; ++i) {
             int j = p_180624_2_.nextInt(16) + 8;
             int k = p_180624_2_.nextInt(16) + 8;
             int l = p_180624_2_.nextInt(worldIn.getHorizon(p_180624_3_.add(j, 0, k)).getY() + 32);
@@ -54,41 +54,28 @@
         super.decorate(worldIn, p_180624_2_, p_180624_3_);
     }
 
-    public static class Mutated extends BiomeGenMutated
-        {
-            private static final String __OBFID = "CL_00000183";
+    public static class Mutated extends BiomeGenMutated {
 
-            public Mutated(int p_i45382_1_, BiomeGenBase p_i45382_2_)
-            {
-                super(p_i45382_1_, p_i45382_2_);
-                this.theBiomeDecorator.treesPerChunk = 2;
-                this.theBiomeDecorator.flowersPerChunk = 2;
-                this.theBiomeDecorator.grassPerChunk = 5;
-            }
+        private static final String __OBFID = "CL_00000183";
 
-            public void genTerrainBlocks(World worldIn, Random p_180622_2_, ChunkPrimer p_180622_3_, int p_180622_4_, int p_180622_5_, double p_180622_6_)
-            {
-                this.topBlock = Blocks.grass.getDefaultState();
-                this.fillerBlock = Blocks.dirt.getDefaultState();
+        public Mutated(int p_i45382_1_, BiomeGenBase p_i45382_2_) {
+            super(p_i45382_1_, p_i45382_2_);
+            this.theBiomeDecorator.treesPerChunk = 2;
+            this.theBiomeDecorator.flowersPerChunk = 2;
+            this.theBiomeDecorator.grassPerChunk = 5;
+        }
 
-                if (p_180622_6_ > 1.75D)
-                {
-                    this.topBlock = Blocks.stone.getDefaultState();
-                    this.fillerBlock = Blocks.stone.getDefaultState();
-                }
-                else if (p_180622_6_ > -0.5D)
-                {
-                    this.topBlock = Blocks.dirt.getDefaultState().withProperty(BlockDirt.VARIANT, BlockDirt.DirtType.COARSE_DIRT);
-                }
+        public void genTerrainBlocks(World worldIn, Random p_180622_2_, ChunkPrimer p_180622_3_, int p_180622_4_, int p_180622_5_, double p_180622_6_) {
+            this.topBlock = Blocks.grass.getDefaultState();
+            this.fillerBlock = Blocks.dirt.getDefaultState();
 
-                this.generateBiomeTerrain(worldIn, p_180622_2_, p_180622_3_, p_180622_4_, p_180622_5_, p_180622_6_);
+            if (p_180622_6_ > 1.75D) {
+                this.topBlock = Blocks.stone.getDefaultState();
+                this.fillerBlock = Blocks.stone.getDefaultState();
+            } else if (p_180622_6_ > -0.5D) {
+                this.topBlock = Blocks.dirt.getDefaultState().withProperty(BlockDirt.VARIANT, BlockDirt.DirtType.COARSE_DIRT);
             }
 
-            public void decorate(World worldIn, Random p_180624_2_, BlockPos p_180624_3_)
-            {
-                this.theBiomeDecorator.decorate(worldIn, p_180624_2_, this, p_180624_3_);
-            }
+            this.generateBiomeTerrain(worldIn, p_180622_2_, p_180622_3_, p_180622_4_, p_180622_5_, p_180622_6_);
         }
-}
+
+        public void decorate(World worldIn, Random p_180624_2_, BlockPos p_180624_3_) {
+            this.theBiomeDecorator.decorate(worldIn, p_180624_2_, this, p_180624_3_);
+        }
+    }
+}
